### ISSUE TITLE
beesd script can't handle BEESHOME on a non-btrfs filesystem

### ISSUE THREAD

**Issue by nitroxis** [Zygo/bees#272]
Hi,

currently, the `beesd` helper script always tries to create `$BEESHOME` as a subvolume (or checks whether it is one). However, it should be fine to have `$BEESHOME` set to e.g. a ext4 root partition, while bees is deduping a separate btrfs HDD, or am I mistaken?

**Comment by nitroxis** [comment]
Proposal for changing the check:

```bash
BEESHOME_FSTYPE=$(stat -f -c %T "$BEESHOME")
if [ "$BEESHOME_FSTYPE" = "btrfs" ]; then
    if [ ! -d "$BEESHOME" ]; then
        INFO "Create subvol $BEESHOME for store bees data"
        btrfs sub cre "$BEESHOME"
    else
        btrfs sub show "$BEESHOME" &> /dev/null || ERRO "$BEESHOME MUST BE A SUBVOL!"
    fi
else
    mkdir -p "$BEESHOME"
fi
```

**Comment by kakra** [comment]
Yeah, it should be completely fine running the hash file on non-btrfs filesystems. I'm running it the same way here - actually by NOT using the beesd wrapper. Maybe create a PR so it gets merged.

**Comment by Connor22** [comment]
Adding onto the script change, I also needed to make sure that the script had access to the volume (such as adding 
`ReadWritePaths=/var/.beeshome` underneath the `[service]` header of the systemd service file) to get it working

**Comment by kakra** [comment]
If you are using customized configs (like a beeshome path differing from defaults), you should create a systemd override file instead to add the needed configs tp the service. You can use `systemctl edit beesd@.service` (or whatever your service is called) to create per-service or even per-instance settings. This way, your individual customizations won't be overwritten by updates.

I'm pretty sure `/var/.beeshome` is a non-default setting.

**Comment by Zygo** [comment]
Currently, `beesd` will:
1. check to see if `$BEESHOME` exists
2. create it as a subvol if not
3. ...end of list

For the default `$BEESHOME` that's a reasonable approach, since the default `$BEESHOME` is on the filesystem where bees runs.  bees absolutely does not run on non-btrfs filesystems, so it's always possible (for root) to create a subvol there.

For non-default `$BEESHOME`, my preferences are:
1.  not do anything to set up `$BEESHOME`.  You're not using the default, you have to RTFM and set it up correctly.
2. attempt to create a subvol, and if that fails, attempt to create a directory without a path, i.e. `mkdir` not `mkdir -p`.

Preference 2 allows creating `$BEESHOME` dynamically, e.g. during installation, whoever set this up created `/var/lib/bees`, and inside that directory there's a `$BEESHOME` directory for every filesystem uuid.

If the user typos `/var/libb/bees`, we shouldn't create `/var/libb` for them--in that case, we should fail, and allow the user the opportunity to correct their typo.
